src/main.cpp

```diff
@@ -181,43 +119,72 @@ int main(int argc, char* argv[])
     }
 #endif // _MSC_VER && _DEBUG
     spDebugConsole = nullptr;
-    unsigned int startupAction = 0;
 
     // due to a Qt bug, this only safely works for both non- and HiDPI displays on 5.12+
     // 5.6 - 5.11 make the application blow up in size on non-HiDPI displays
 #if defined (Q_OS_UNIX) && (QT_VERSION >= QT_VERSION_CHECK(5, 12, 0))
     QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
 #endif
 
-    QScopedPointer<QCoreApplication> initApp(createApplication(argc, argv, startupAction));
-    auto * app = qobject_cast<QApplication*>(initApp.data());
+#if defined(Q_OS_MACOS) && (QT_VERSION < QT_VERSION_CHECK(5, 12, 0))
+        // Workaround for horrible mac rendering issues once the mapper widget
+        // is open - see https://bugreports.qt.io/browse/QTBUG-41257
+        QApplication::setAttribute(Qt::AA_DontCreateNativeWidgetSiblings);
+#elif defined(Q_OS_FREEBSD)
+        // Cure for diagnostic:
+        // "Qt WebEngine seems to be initialized from a plugin. Please set
+        // Qt::AA_ShareOpenGLContexts using QCoreApplication::setAttribute
+        // before constructing QGuiApplication."
+        QApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
+#endif
+
+    auto * app = qobject_cast<QApplication*>(new QApplication(argc, argv));
+
+#if defined(Q_OS_WIN32) && defined(INCLUDE_UPDATER)
+    auto abortLaunch = runUpdate();
+    if (abortLaunch) {
+        return 0;
+    }
+#endif
+
+    // Turn the cursor into the waiting one during startup, so something shows
+    // activity even if the quiet, no splashscreen startup has been used
+    app->setOverrideCursor(QCursor(Qt::WaitCursor));
+    app->setOrganizationName(QStringLiteral("Mudlet"));
+
+    if (mudlet::scmIsPublicTestVersion) {
+        app->setApplicationName(QStringLiteral("Mudlet Public Test Build"));
+        app->setApplicationVersion(APP_VERSION APP_BUILD);
+    } else {
+        app->setApplicationName(QStringLiteral("Mudlet"));
+        app->setApplicationVersion(APP_VERSION);
+    }
+
+    QCommandLineParser parser;
+    QCommandLineOption profileToOpen("profile", QCoreApplication::translate("main", "Profile to open automatically"), QCoreApplication::translate("main", "profile"));
+    parser.addOption(profileToOpen);
+
+    QCommandLineOption showHelp(QStringList() << "h" <<"help", QCoreApplication::translate("main", "Display help and exit"));
```

Comment:
Seemingly two different ways of constructing `QCommandLineOption` are used?

Reply:
The first defines an option with a value that only has a long form (--profile), the second defines an option that takes two forms (-h or --help) but does not also take a value. Which is why the second has the QStringList for the first argument and the first has the extra argument at the end, for what the value is called. I believe it gets used primarily if you invoke the --help handler provided by QCommandLineParser